### PR TITLE
Mention BigInt.prototype.toJSON

### PR DESCRIPTION
**Pull request by Josh-Cena**
#### Motivation

This brings back the information removed in https://github.com/mdn/content/pull/14763.

`BigInt.prototype.toJSON` is actually not discouraged at all — it's very much a language design to let you provide your own hook. See the long thread at https://github.com/tc39/proposal-bigint/issues/24.

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/bigint/index.md [hunk 1]
```diff
@@ -233,31 +233,46 @@ The operations supported on BigInt values are not constant-time, and are thus op
 
 ### Use within JSON
 
-Using [`JSON.stringify()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify) with any BigInt value will raise a `TypeError`, as BigInt values aren't serialized in JSON by default. However, you can use the [replacer](/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#the_replacer_parameter) parameter of `JSON.stringify` to serialize BigInt properties without error:
+Using [`JSON.stringify()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify) with any BigInt value will raise a `TypeError`, as BigInt values aren't serialized in JSON by default. However, `JSON.stringify()` specifically leaves a backdoor for BigInt values: it would try to call the BigInt's `toJSON()` method. (It doesn't do so for any other primitive values.) Therefore, you can implement your own `toJSON()` method (which is one of the few cases where monkeypatching is not explicitly discouraged):
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/bigint/index.md:236]: _monkeypatching_ is not an obvious term. It should link to a glossary entry or replaced by something simpler.

File: files/en-us/web/javascript/reference/global_objects/bigint/index.md [hunk 2]
```diff
@@ -233,31 +233,46 @@ The operations supported on BigInt values are not constant-time, and are thus op
...
+
+If you do not wish to monkeypatch `BigInt.prototype`, you can use the [`replacer`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#the_replacer_parameter) parameter of `JSON.stringify` to serialize BigInt values:
 
 ```js
 function replacer(key, value) {
-  if (key === 'big') {
+  if (key === "big") {
     return value.toString();
   }
   return value;
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/bigint/index.md:258]:
```suggestion
  return key === "big" ? value.toString() : value;
```

File: files/en-us/web/javascript/reference/global_objects/bigint/index.md [hunk 3]
```diff
@@ -233,31 +233,46 @@ The operations supported on BigInt values are not constant-time, and are thus op
...
-If you have JSON data containing values you know will be large integers, you can use the [reviver](/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#using_the_reviver_parameter) parameter of `JSON.parse` to handle them:
+If you have JSON data containing values you know will be large integers, you can use the [`reviver`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#using_the_reviver_parameter) parameter of `JSON.parse` to handle them:
 
 ```js
 function reviver(key, value) {
-  if (key === 'big') {
+  if (key === "big") {
     return BigInt(value);
   }
   return value;
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/bigint/index.md:278]:
```suggestion
  return key === "big" ? BigInt(value) : value;
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]

**Comment by teoli2003** [comment]
I've also added @sideshowbarker as a reviewer, as it reviewed the original change and may have additional insight.

**Review by sideshowbarker — APPROVED** [review]
Thanks — TIL; looks good to me

**Review by teoli2003 — APPROVED** [review]
👍

**Pull request merged**